tests/system_metrics/test_system_metrics_logging.py

```diff
@@ -17,7 +17,16 @@ def disable_system_metrics_logging():
     mlflow.set_system_metrics_node_id(None)
 
 
+def wait_for_condition(condition_func, timeout=10, check_interval=1):
+    for _ in range(timeout // check_interval):
+        if condition_func():
+            return True
```

Comment:
Do we need to return True/False?

Reply:
definitely. Updated to make this cleaner